src/main/java/com/microsoft/azure/relay/TokenRenewer.java

```diff
@@ -61,23 +68,38 @@ void onRenewTimer() {
 	}
 
 	private void scheduleRenewTimer(SecurityToken token) {
-		this.renewTimer = new Timer();
-		Duration interval = Duration.between(Instant.now(), token.getExpiresAtUtc());
-		if (interval.isNegative()) {
-			RelayLogger.logEvent("tokenRenewNegativeDuration", this.listener);
-			return;
-		}
+	    if (this.renewTimer == null) {
+	        this.renewTimer = new Timer();    
+	    }
```

Comment:
Is it preferable to have this outside the scheduleLock protected code? Would it be better to put this inside the lock?

Reply:
Don't think it would make a difference in this case in or out of the locked block

---
In reply to: [338832149](https://github.com/Azure/azure-relay-java/pull/54#discussion_r338832149) [](ancestors = 338832149)

Reply:
Please put it inside the block.

---
In reply to: [338835499](https://github.com/Azure/azure-relay-java/pull/54#discussion_r338835499) [](ancestors = 338835499,338832149)